gqlalchemy/instance_runner.py

```diff
@@ -0,0 +1,233 @@
+# Copyright (c) 2016-2022 Memgraph Ltd. [https://memgraph.com]
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import docker
+import os
+import psutil
+import socket
+import subprocess
+import time
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Any, Dict, Optional, Union
+from .memgraph import Memgraph
+
+
+DEFAULT_MEMGRAPH_PATH = "/usr/lib/memgraph/memgraph"
+
+TIMEOUT_ERROR_MESSAGE = "Waited too long for the port {port} on host {host} to start accepting connections."
+DOCKER_TIMEOUT_ETTOT_MESSAGE = "Waited too long for the Docker container to start."
+MEMGRAPH_CONNECTION_ERROR_MESSAGE = "The Memgraph process probably died."
+
+
+class DockerImage(Enum):
+    MEMGRAPH = "memgraph/memgraph"
+    MAGE = "memgraph/memgraph-mage"
+
+
+def wait_for_port(host: str = "127.0.0.1", port: int = 7687, delay: float = 0.01, timeout: float = 5.0) -> None:
```

Comment:
All strings should be promoted to constants, including "127.0.0.1"

Reply:
Fixed